src/components/ContactList.tsx

```diff
@@ -37,20 +47,66 @@ const ContactList = () => {
         fetchContacts();
     }, []) 
 
+    //TODO: optimize search so that it can find partial match to name search
+    //TODO: fix the any typing in matchByNameSearResults
+    const getMatchingNames = (nameStr: string | null) => {
+        const namesArray: any[] = [];
+        contacts.map((contact: any) => {
+            if (contact.name === nameStr) {
```

Comment:
To search for substrings:

```suggestion
            if (contact.name.includes(nameStr)) {
```

Reply:
Also, why not also check other address fields? Or will you do that in a later PR?

Reply:
I was trying to get the MVP version going and then optimize it later. Ideally it would search for more than just the name.

Reply:
Also, I forgot that .includes() would search for substrings! Exactly what I wanted!